presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java

```diff
@@ -171,10 +171,14 @@ public void stop()
                 .with(timeout, retry)
                 .with(executorService);
 
-        ImmutableList.copyOf(containers.values())
-                .stream()
-                .filter(DockerContainer::isRunning)
-                .forEach(container -> executor.run(container::tryStop));
+        for (DockerContainer container : ImmutableList.copyOf(containers.values())) {
+            try {
+                executor.runAsync(container::tryStop).get();
+            }
+            catch (InterruptedException | ExecutionException e) {
+                log.warn("Could not stop container %s due to %s", container.getLogicalName(), e);
```

Comment:
Is it intentional to not log the stack trace?

Reply:
yes

Reply:
please add explanatory comment